gbpservice/nfp/configurator/drivers/firewall/asav/asav_fw_driver.py

```diff
@@ -0,0 +1,960 @@
+        try:
+            provider_interface_position = self.get_interface_position(
+                                                        mgmt_ip, provider_mac)
+
+            commands = []
+            interface_id = self._get_asav_interface_id(
+                                                provider_interface_position)
+            commands.append("interface " + interface_id)
+            commands.append("no policy-route route-map pbrmap" +
+                            source_cidr.replace('/', '_'))
+            commands.append("no route-map pbrmap" + source_cidr.replace(
+                '/', '_'))
+            commands.append("clear configure access-list pbracl" +
+                            source_cidr.replace('/', '_'))
+            commands.append("clear configure interface " + interface_id)
+
+            self.configure_bulk_cli(mgmt_ip, commands)
+        except Exception as err:
+            msg = ("Exception while deleting pbr route. "
+                   "commands: %s, Reason: %s" % (commands, err))
+            LOG.error(msg)
+            return msg
+        else:
+            return const.STATUS_SUCCESS
+
+    @staticmethod
+    def get_asav_mac(mac_addr):
+        """ Converts standard MAC address to ASAv format.
+
+        :param mac_addr: MAC address
+
+        Returns: ASAv MAC address.
+
+        """
+
+        if not mac_addr:
+            raise Exception('Get asav mac received an empty mac address')
+        l = mac_addr.split(':')
+        asav_mac = ""
+        for i in range(0, len(l), 2):
+            asav_mac += (l[i] + l[i+1] + ".")
+
+        return asav_mac[:-1]
+
+    @staticmethod
+    def get_asav_macs(mac_list):
+        """ Converts standard MAC address to ASAv format.
+
+        :param mac_addr: A list of MAC addresses
+
+        Returns: A tuples of ASAv MAC addresses.
+
+        """
+        asav_mac_list = list()
+        for mac_addr in mac_list:
+            if not mac_addr:
+                asav_mac_list.append(None)
+                continue
+            l = mac_addr.split(':')
+            asav_mac = ""
+            for i in range(0, len(l), 2):
+                asav_mac += (l[i] + l[i+1] + ".")
+            asav_mac_list.append(asav_mac[:-1])
+
+        return tuple(asav_mac_list)
+
+""" Firewall as a service driver for handling firewall
+service configuration requests.
+
+We initialize service type and service vendor in this class because
+agent loads class object only for those driver classes that have service
+type and service vendor as class attributes. Also, only this driver
+class is exposed to the agent.
+
+"""
+
+
+class FwaasDriver(FwGenericConfigDriver):
+    service_type = const.SERVICE_TYPE
+    service_vendor = const.ASAV
+
+    def __init__(self, conf):
+        self.conf = conf
+        self.register_config_options()
+        self.timeout = const.REST_TIMEOUT
+        self.rest_api = RestApi(self.timeout)
+        self.port = const.ASAV_CONFIGURATION_SERVER_PORT
```

Comment:
These are ASAV specific constants and should not be in common firewall constants.